packages/foam-vscode/src/features/placeholders.ts

```diff
@@ -0,0 +1,85 @@
+import * as vscode from 'vscode';
+import { Foam, Resource, isNote, isPlaceholder, isAttachment } from 'foam-core';
+import {
+  FilteredResourcesConfigGroupBy,
+  getPlaceholdersConfig,
+} from '../settings';
+import { FoamFeature } from '../types';
+import { FilteredResourcesProvider } from './filtered-resources';
+
+const feature: FoamFeature = {
+  activate: async (
+    context: vscode.ExtensionContext,
+    foamPromise: Promise<Foam>
+  ) => {
+    const foam = await foamPromise;
+    const workspacesURIs = vscode.workspace.workspaceFolders.map(
+      dir => dir.uri
+    );
+    const provider = new FilteredResourcesProvider(
+      foam.workspace,
+      foam.services.dataStore,
+      'placeholders',
+      'placeholder',
+      isPlaceholderResource,
+      {
+        ...getPlaceholdersConfig(),
+        workspacesURIs,
+      }
+    );
+
+    context.subscriptions.push(
+      vscode.window.registerTreeDataProvider(
+        'foam-vscode.placeholders',
+        provider
+      ),
+      vscode.commands.registerCommand(
+        'foam-vscode.group-placeholders-by-folder',
+        () => provider.setGroupBy(FilteredResourcesConfigGroupBy.Folder)
+      ),
+      vscode.commands.registerCommand(
+        'foam-vscode.group-placeholders-off',
+        () => provider.setGroupBy(FilteredResourcesConfigGroupBy.Off)
+      ),
+      foam.workspace.onDidAdd(() => {
+        provider.refresh();
+        foam.workspace.resolveLinks();
```

Comment:
is there a reason for calling this? I don't believe you need to resolve the links here

Reply:
If we don't resolve the links, then placeholders do not get updated. This is an issue when we click a true "placeholder" in the explorer view, which creates a blank note for that "placeholder". The blank note then shows up, but the "placeholder" link is still present. Resolving links cleans up the placeholders.

Reply:
can you check now whether this is still needed? 
I believe the behavior you are describing is due to a bug in 0.10.0 that was fixed in 0.10.1

Reply:
Yes, pulling in 0.10.1 seems to have resolved this.